### ISSUE TITLE
Decoder stopped with 'max_decoder_steps，synthesized wave file is unexpected long

### ISSUE THREAD

**Issue by ryantan1** [mozilla/TTS#483]
Hi erogol，

appreciate for your great effort to fix #471  issue on dev branch.
I have pulled the code, and happy to see it is still runing with so many epochs

meanwhile I got another issue, after 1405 steps, I tried to synthesize wav with certain sentence from training dataset.
it takes really long time, display the message "Decoder stopped with max_decoder_steps", the generated wav file is 58s  long while original wav is just about 2s.  

simply look into code:

it happened at tts/layers/tacotron2.py inference below, it seems inputs.shape[0] always equals to 1 
so the problem is caused by stop_token which seems predicted by stopnet.  please help, thanks a lot~

---------------------------------
#print(f"---- stop_tokens:{stop_tokens}\n t:{t} inputs.shape:{inputs.shape}\n")
            if stop_token > self.stop_threshold and t > inputs.shape[0] // 2:
                break
            if len(outputs) == self.max_decoder_steps:
                print("   | > Decoder stopped with 'max_decoder_steps")
                break

---------------------------------

test environment:
UBUNTU 1804, CPU 8 core, MEMORY 16GB, no GPU
test branch: dev
commit: e0f26d3

-----some logs from terminal: -----
rary libcudart.so.10.1
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > num_mels:80
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:True
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:8000.0
 | > spec_gain:20.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:60
 | > do_sound_norm:False
 | > stats_path:None
 | > hop_length:256
 | > win_length:1024
 > Using model: Tacotron2
 > Text: deng4 xiao3 ping2 yu3 sa4 qie4 er3 hui4 wu4
   **| > Decoder stopped with 'max_decoder_steps**
 > Run-time: 133.1087965965271
 > Real-time factor: 2.2934655696626844
 > Time per step: 0.00010401204826057279
 > Saving output to ./deng4_xiao3_ping2_yu3_sa4_qie4_er3_hui4_wu4.wav

**Comment by ryantan1** [comment]
it seems my screenshot image can't be uploaded, the tensorboard shows stopnet loss keeps down, now it is 0.5609, does it link to the stoptoken calculation in decoder.inference? what should the good stopnet loss be？

avg_decoder_loss keeps up， is it normal? thanks~

**Comment by ryantan1** [comment]
the evaluation has some proplem during training and build 58 s long wav test file

**Comment by SanjaESC** [comment]
1405 are far too few steps. Try training to at least 10.000 steps and check again.

**Comment by ryantan1** [comment]
@SanJaESC，thanks your help, ou, that means i have to wait for several days. today I just got GPU test environment, that maybe short the training time, but torch.cuda.is_available() get False. while use nvidia-smi -q to check, I got the following info:

**Comment by ryantan1** [comment]
so it seems GPU is there, could you please guide me how to let torch to use it? thank~

**Comment by ryantan1** [comment]
I can use the GPU now after install cuda torch version correctly
